source/lexbor/dom/interfaces/element.c

```diff
@@ -1095,63 +1096,63 @@ static bool
 lxb_dom_elements_by_attr_cmp_full(lxb_dom_element_cb_ctx_t *ctx,
                                   lxb_dom_attr_t *attr)
 {
-    if (ctx->value_length == attr->value->length
+    if (attr->value && ctx->value_length == attr->value->length
```

Comment:
@phoerious 

```C
if (attr->value && 
```
replace
```C
if (attr->value != NULL && 
```

Only boolean variables are not checked for specific values.

Reply:
I wasn't quite sure what you coding style here is. Usually, checking explicitly for == or != `NULL` is an antipattern and your code base wasn't entirely consistent either, but I'll change it.